benchbuild/projects/benchbuild/ruby.py

```diff
@@ -28,17 +26,20 @@ def compile(self):
         with local.cwd(unpack_dir):
             with local.env(CC=str(clang), CXX=str(clang_cxx)):
                 configure = local["./configure"]
-                run.run(configure["--with-static-linked-ext",
-                                  "--disable-shared"])
-            run.run(make["-j", get_number_of_jobs(CFG)])
+                configure = run.watch(configure)
+                configure("--with-static-linked-ext", "--disable-shared")
+            make_ = run.watch(make)
+            make_("-j", get_number_of_jobs(CFG))
 
-    def run_tests(self, runner):
+    def run_tests(self):
         unpack_dir = local.path('ruby-{0}'.format(self.version))
         ruby_n = wrapping.wrap(unpack_dir / "ruby", self)
         testdir = local.path(self.testdir)
 
         with local.env(RUBYOPT=""):
-            run.run(ruby[testdir / "benchmark" / "run.rb", "--ruby=\"" +
-                         str(ruby_n) + "\"", "--opts=\"-I" + testdir / "lib" +
-                         " -I" + testdir / "." + " -I" +
-                         testdir / ".ext" / "common" + "\"", "-r"])
+            ruby_ = run.watch(ruby)
+            ruby(
+                testdir / "benchmark" / "run.rb",
+                "--ruby=\"" + str(ruby_n) + "\"",
+                "--opts=\"-I" + testdir / "lib" + " -I" + testdir / "." +
+                " -I" + testdir / ".ext" / "common" + "\"", "-r")
```

Comment:
For some of those big strings a simple f-string would be much nicer :)

Reply:
Fixed in cc611b4a00e299b1eed8084f79553c9b3259e800